Upstream packaging.diff

ckamm opened the pull request:
Please review and test. The admin removal looks dubious to me! See #5957

dschmidt commented on CMakeLists.txt:
```diff
@@ -243,7 +267,7 @@ configure_file(version.h.in ${CMAKE_CURRENT_BINARY_DIR}/version.h)
     install(FILES sync-exclude.lst DESTINATION ${OWNCLOUD_OSX_BUNDLE}/Contents/Resources/)
     configure_file(sync-exclude.lst bin/${OWNCLOUD_OSX_BUNDLE}/Contents/Resources/sync-exclude.lst COPYONLY)
-else()
+elseif(BUILD_CLIENT)
```
This is nonsense - don't know how it got into the diff

ckamm replied:
@dschmidt Why is it nonsense? Shouldn't the whole preparation for sync-exclude.lst be done for BUILD_CLIENT only?

dschmidt commented on CMakeLists.txt:
```diff
@@ -247,7 +247,6 @@ if(BUILD_CLIENT)
     if(NOT BUILD_LIBRARIES_ONLY)
         add_subdirectory(doc)
         add_subdirectory(doc/dev)
-        add_subdirectory(admin)
```
Was this really in the diff? Looks like an accident or something - it looks pretty wrong, don't even know what it would be good for in the linux packages

dschmidt commented:
I took the liberty of adding the RPATH change to your branch/PR, hope that was ok :trollface:

ogoffart commented on src/libsync/theme.cpp:
```diff
@@ -505,5 +505,20 @@ QString Theme::oauthClientSecret() const
+    stream << "Git revision " << GIT_SHA1 << endl;
+#endif
+    stream << "Using Qt " << qVersion() << ", built against Qt " << QT_VERSION_STR << endl;
+    stream << "Using '" << QSslSocket::sslLibraryVersionString() << "'" << endl;
```
I checked,  it should not crash.

dschmidt replied:
What's the output then?

ogoffart replied:
nothing:  https://code.woboq.org/qt5/qtbase/src/network/ssl/qsslsocket_openssl11.cpp.html#180

arrguably, it is a bit strange to have an empty Using,  but this should not happen in practice.

ogoffart commented on src/cmd/cmd.cpp:
```diff
@@ -196,8 +196,7 @@ void help()
 {
-    const char *binaryName = APPLICATION_EXECUTABLE "cmd";
-    std::cout << binaryName << " version " << qPrintable(Theme::instance()->version()) << std::endl;
+    std::cout << qPrintable(Theme::instance()->versionSwitchOutput());
```
-1:   Encoding!  should be qPrintableUtf8
(the application executable might contains non-ascii)

ogoffart commented on cmake/modules/DefineCompilerFlags.cmake:
```diff
@@ -16,7 +16,8 @@ if (${CMAKE_C_COMPILER_ID} MATCHES "(GNU|Clang)")
     # FIXME Can we somehow not use those flags for sqlite3.* but use them for the rest of csync?
     if (NOT USE_OUR_OWN_SQLITE3)
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -pedantic -pedantic-errors")
+        # -pedantic-errors explodes in src/csync/csync_private.h:166
```
csync_private.h:166   is a comment for me.
So I guess this comment is already outdated and should be removed.

Maybe pedentic-errors would make sens on a CI, but not when building packages, so it is fine to remove it.

dschmidt replied:
Ooh of course, it was early, when I wrote the comment ... 😄 
Yeah, let's just remove it.

ogoffart commented on CMakeLists.txt:
```diff
@@ -162,28 +177,32 @@ if(APPLE)
-  find_package(Sparkle)
-endif(APPLE)
+if(BUILD_CLIENT)
+    find_package(OpenSSL 1.0.0 REQUIRED)	# still needed?
```
-1:  Not needed

ckamm commented:
@dschmidt Sure :)

ckamm commented:
@ogoffart @dschmidt I did the `qPrintableUtf8` change, removed the compile flags change, removed the unneeded `find_package(OpenSSL)`.

Open issue: Is the `admin` directory intentionally not traversed in cmakelists? @guruz maybe?

guruz commented:
> Open issue: Is the admin directory intentionally not traversed in cmakelists? @guruz maybe?

👎  

Indeed, because for macOS it contains important stuff like ```macdeployqt.py```. So it needs to be there actually.

The packaging.diff was applied on Linux only.

@jnweiger opinion?

guruz commented:
FYI @hefee @dragotin

guruz commented:
> But removing the admin folder via de8ab35 is probably not needed

ack

> But removing the admin folder via de8ab35 is probably not needed. I suggest change clean_tarball.sh to not remove the admin folder.

clean_tarball.sh was only intended for external linux packagers, not for us doing all kind of stuff

> I cannot currently run a test with the nightlies, looks like part of the changes are already in master..

Wait for this PR to get merged and then see that packaging.diff should not be needed anymore?

ckamm commented:
@jnweiger @guruz I've removed the commit that skipped the admin/ directory. I'll merge this now. Let's test and get this rolling :)

ogoffart commented on src/libsync/theme.cpp:
```diff
@@ -505,5 +505,20 @@ QString Theme::oauthClientSecret() const
+{
+    QString helpText;
+    QTextStream stream(&helpText);
+    stream << appName().toLatin1().constData()
```
Ooops!  Encoding  (not new in this patch)
